### PR TITLE
[Review] Using clang-format to format code

### PR DESCRIPTION
**Pull request by tcahuzax**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/319%23issuecomment-163186769%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23issuecomment-163270521%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23issuecomment-163334818%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23issuecomment-163335457%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46933755%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46933805%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934050%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934059%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934301%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934476%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46962510%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46962594%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r47062208%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r47062285%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r47062372%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23issuecomment-163186769%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%5Cr%5Cn%40dawagner%20%20%40krocard%20%20Plz%20review%22%2C%20%22created_at%22%3A%20%222015-12-09T10%3A49%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-09T14%3A48%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%5Cr%5Cn%40dawagner%20%40krocard%20Plz%20review%22%2C%20%22created_at%22%3A%20%222015-12-09T17%3A37%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-12-09T17%3A40%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20bindings/c/Test.cpp%2085%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934050%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20is%20this%20put%20on%20a%20separate%20line%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T10%3A00%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20agree%20but%20I%20don%27t%20know%20to%20avoid%20it.%20I%20think%20that%20clang-format%20doesn%27t%20like%20multiline%20strings.%22%2C%20%22created_at%22%3A%20%222015-12-09T08%3A32%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/Test.cpp%3AL135-196%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20bindings/c/Test.cpp%2093%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934059%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22ditto%22%2C%20%22created_at%22%3A%20%222015-12-08T10%3A00%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/Test.cpp%3AL135-196%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20parameter/AreaConfiguration.cpp%20124%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934301%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20get%20that%20we%20want%20this%20for%20methods%20defined%20in%20headers%20%28for%20concision%29%20but%20it%20is%20against%20our%20coding%20style%20to%20do%20it%20in%20source%20files.%20Is%20there%20a%20way%20to%20selectively%20turn%20it%20on%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T10%3A03%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22No%20it%27s%20not%20possible%20to%20discriminate%20headers/source.%20So%20we%20have%20to%20choose.%22%2C%20%22created_at%22%3A%20%222015-12-09T08%3A30%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/AreaConfiguration.cpp%3AL128-174%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20bindings/c/ParameterFramework.h%20105%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46933805%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20isn%27t%20this%20particular%20asterisk%20correctly%20aligned%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T09%3A58%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222015-12-08T14%3A59%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/ParameterFramework.h%3AL192-200%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20parameter/ArrayParameter.cpp%20291%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46934476%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Wow%2C%20this%20line%20break%20is%20really%20strange.%22%2C%20%22created_at%22%3A%20%222015-12-08T10%3A05%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20agree%2C%20I%20don%27t%20have%20found%20an%20option%20to%20avoid%20it.%22%2C%20%22created_at%22%3A%20%222015-12-09T08%3A31%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ArrayParameter.cpp%3AL426-444%22%7D%2C%20%22Pull%20790249c98931418ce8b43e4bb6eb9e451948cac3%20bindings/c/ParameterFramework.h%2078%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/319%23discussion_r46933755%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Wrongfully%20aligned%20comment.%22%2C%20%22created_at%22%3A%20%222015-12-08T09%3A57%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222015-12-08T14%3A59%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/ParameterFramework.h%3AL113-138%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/dawagner%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/dawagner'></a><a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 bindings/c/ParameterFramework.h 78'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46933755'>File: bindings/c/ParameterFramework.h:L113-138</a></b>
- <a href='https://github.com/dawagner'></a> Wrongfully aligned comment.
- <a href='https://github.com/tcahuzax'></a> fixed
- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 bindings/c/ParameterFramework.h 105'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46933805'>File: bindings/c/ParameterFramework.h:L192-200</a></b>
- <a href='https://github.com/dawagner'></a> Why isn't this particular asterisk correctly aligned?
- <a href='https://github.com/tcahuzax'></a> fixed
- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 bindings/c/Test.cpp 85'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46934050'>File: bindings/c/Test.cpp:L135-196</a></b>
- <a href='https://github.com/dawagner'></a> Why is this put on a separate line?
- <a href='https://github.com/tcahuzax'></a> I agree but I don't know to avoid it. I think that clang-format doesn't like multiline strings.
- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 bindings/c/Test.cpp 93'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46934059'>File: bindings/c/Test.cpp:L135-196</a></b>
- <a href='https://github.com/dawagner'></a> ditto
- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 parameter/AreaConfiguration.cpp 124'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46934301'>File: parameter/AreaConfiguration.cpp:L128-174</a></b>
- <a href='https://github.com/dawagner'></a> I get that we want this for methods defined in headers (for concision) but it is against our coding style to do it in source files. Is there a way to selectively turn it on?
- <a href='https://github.com/tcahuzax'></a> No it's not possible to discriminate headers/source. So we have to choose.
- [ ] <a href='#crh-comment-Pull 790249c98931418ce8b43e4bb6eb9e451948cac3 parameter/ArrayParameter.cpp 291'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#discussion_r46934476'>File: parameter/ArrayParameter.cpp:L426-444</a></b>
- <a href='https://github.com/dawagner'></a> Wow, this line break is really strange.
- <a href='https://github.com/tcahuzax'></a> I agree, I don't have found an option to avoid it.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/319#issuecomment-163186769'>General Comment</a></b>
- <a href='https://github.com/tcahuzax'></a> :+1:
@dawagner  @krocard  Plz review
- <a href='https://github.com/krocard'></a> :shipit:
- <a href='https://github.com/tcahuzax'></a> :+1:
@dawagner @krocard Plz review

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/319?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/319?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/319?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/319'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/319'>Refresh</a>.<a href='#crh-update-none'></a>

File: bindings/c/ParameterFramework.h [hunk 1]
```diff
@@ -113,25 +113,25 @@ typedef struct {
...
+                           * @example { "Red", "Green", "Blue", NULL }
+                           *
+                           * For an exclusive criterion, the list must not contain more elements then
+                           *                             INT_MAX.
+                           * For an inclusive criterion, the list must not contain more elements then
+                           *                             sizeof(int) * BIT_CHAR - 1.
+                           *                             Ie: (int)1 << n must *not* overflow (UB),
+                           *                                 were n is the number of element in the
+                           *                                 list. @see pfwSetCriterion
+                           */
```
Comment (dawagner) [inline bindings/c/ParameterFramework.h]: Wrongfully aligned comment.

File: bindings/c/ParameterFramework.h [hunk 2]
```diff
@@ -190,7 +192,8 @@ CPARAMETER_EXPORT
 bool pfwSetCriterion(PfwHandler *handle, const char name[], int value) NONNULL USERESULT;
 /** Get a criterion value given its name.
   * Same usage as pfwSetCriterion except that value is an out param.
-  * Get criterion will return the last value setted with pfwSetCriterion independantly of pfwCommitCritenio.
+  * Get criterion will return the last value setted with pfwSetCriterion independantly of
+ * pfwCommitCritenio.
```
Comment (dawagner) [inline bindings/c/ParameterFramework.h]: Why isn't this particular asterisk correctly aligned?

File: bindings/c/Test.cpp [hunk 3]
```diff
@@ -137,61 +135,61 @@ TEST_CASE_METHOD(Test, "Parameter-framework c api use") {
         </Subsystem>");
     TmpFile libraries("<?xml version='1.0' encoding='UTF-8'?>\
         <SystemClass Name='test'>\
-            <SubsystemInclude Path='" + system.getPath() + "'/>\
+            <SubsystemInclude Path='" +
+                      system.getPath() + "'/>\
```
Comment (dawagner) [inline bindings/c/Test.cpp]: Why is this put on a separate line.

File: bindings/c/Test.cpp [hunk 4]
```diff
@@ -137,61 +135,61 @@ TEST_CASE_METHOD(Test, "Parameter-framework c api use") {
...
         </SystemClass>");
     TmpFile config("<?xml version='1.0' encoding='UTF-8'?>\
         <ParameterFrameworkConfiguration\
             SystemClassName='test' TuningAllowed='false'>\
             <SubsystemPlugins/>\
-            <StructureDescriptionFileLocation Path='" + libraries.getPath() + "'/>\
+            <StructureDescriptionFileLocation Path='" +
+                   libraries.getPath() + "'/>\
```
Comment (dawagner) [inline bindings/c/Test.cpp]: ditto

File: parameter/AreaConfiguration.cpp [hunk 5]
```diff
@@ -125,52 +128,46 @@ bool CAreaConfiguration::serializeXmlSettings(CXmlElement& xmlConfigurableElemen
...
 
     // Inner becomes valid
     setValid(true);
 }
 
-CParameterBlackboard& CAreaConfiguration::getBlackboard()
-{
-    return _blackboard;
-}
+CParameterBlackboard &CAreaConfiguration::getBlackboard() { return _blackboard; }
```
Comment (dawagner) [inline parameter/AreaConfiguration.cpp]: I get that we want this for methods defined in headers (for concision) but it is against our coding style to do it in source files. Is there a way to selectively turn it on?

File: parameter/ArrayParameter.cpp [hunk 6]
```diff
@@ -399,16 +426,18 @@ bool CArrayParameter::doSet(type value, size_t offset, CParameterAccessContext&
...
     // Read blackboard
-    const CParameterBlackboard* pBlackboard = parameterAccessContext.getParameterBlackboard();
+    const CParameterBlackboard *pBlackboard = parameterAccessContext.getParameterBlackboard();
 
     // Beware this code works on little endian architectures only!
     pBlackboard->readInteger(&uiData, getSize(), offset);
 
-    return static_cast<const CParameterType*>(getTypeElement())->fromBlackboard(value, uiData, parameterAccessContext);
+    return static_cast<const CParameterType *>(getTypeElement())
+        ->fromBlackboard(value, uiData, parameterAccessContext);
```
Comment (dawagner) [inline parameter/ArrayParameter.cpp]: Wow, this line break is really strange.

File: bindings/c/ParameterFramework.h (hunk 1 above)
Comment (tcahuzax) [inline bindings/c/ParameterFramework.h]: fixed

File: bindings/c/ParameterFramework.h (hunk 2 above)
Comment (tcahuzax) [inline bindings/c/ParameterFramework.h]: fixed

File: parameter/AreaConfiguration.cpp (hunk 5 above)
Comment (tcahuzax) [inline parameter/AreaConfiguration.cpp]: No it's not possible to discriminate headers/source. So we have to choose.

File: parameter/ArrayParameter.cpp (hunk 6 above)
Comment (tcahuzax) [inline parameter/ArrayParameter.cpp]: I agree, I don't have found an option to avoid it.

File: bindings/c/Test.cpp (hunk 3 above)
Comment (tcahuzax) [inline bindings/c/Test.cpp]: I agree but I don't know to avoid it. I think that clang-format doesn't like multiline strings.

**Comment by tcahuzax** [comment]
:+1: 
@dawagner  @krocard  Plz review

**Comment by krocard** [comment]
:shipit:

**Comment by tcahuzax** [comment]
:+1: 
@dawagner @krocard Plz review

**Comment by dawagner** [comment]
:shipit:

**Pull request merged**